### PR TITLE
Introduce receive optimisation to dets calls

### PR DESCRIPTION
**Pull request by ssmyczynski**
Current implementation of `dets:req/2` function in `dets` module is not using receive optimisation.

Module `dets` is used as a backend for `mnesia`'s `disc_only_copies` tables. All dirty operations, or symmetric transactions (`sym_trans`) are executed in context of a single `mnesia_tm` process which means that in case of using `disc_only_copies` tables we are actually vulnerable to non-optimised `dets` calls in case when `mnesia_tm` message queue is busy.

The proposed solution introduces receive optimisation to avoid that potential problem.

### CODE REVIEW

File: lib/stdlib/src/dets.erl [hunk 1]
```diff
@@ -1237,15 +1238,27 @@ treq(Tab, R) ->
...
+    Pid;
+pidof(Pid) ->
+    Pid.
+
+%% Inlined.
+resp({Pid, Ref} = _From, Message) ->
+    Pid ! {self(), Ref, Message}, ok;
+resp(Pid, Message) ->
+    Pid ! {self(), Message}, ok.
+
```
Comment (mikpe) [inline lib/stdlib/src/dets.erl:1261]: Do you really need to keep support for replying to old Ref-less calls?

Comment (ssmyczynski) [reply lib/stdlib/src/dets.erl:1258]: You are right, we do not need to keep it anymore. Fixed.

Comment (ssmyczynski) [reply lib/stdlib/src/dets.erl:1258]: `dets` server sends some messages to itself without using references (`auto_save` | `may_grow` | `{delayed_write, _}`) but in that case it does not send nor expect any response.

**Comment by mikpe** [comment]
LGTM

**Comment by garazdawi** [comment]
Thanks!

**Pull request merged**